sdk/core/azure-core/inc/response.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <http/http.hpp>
+
+namespace Azure { namespace Core {
+
+  template <class T> class Response {
+    T m_value;
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    // Require a raw response to create a Response T
+    explicit Response(T const initialValue, std::unique_ptr<Http::RawResponse>&& rawResponse)
+        : m_value(std::move(initialValue)), m_rawResponse(std::move(rawResponse))
+    {
+    }
+
+    // Do not give up raw response ownership.
+    Http::RawResponse& GetRawResponse() { return *this->m_rawResponse; }
+
+    T& operator=(const Response& other) = delete;
+
+    const T* operator->() const { return &this->m_value; };
+    T* operator->() { return &this->m_value; };
+    T& operator*() { return this->m_value; };
+    const T& operator*() const { return this->m_value; };
+
+    T GetValue() { return std::move(this->m_value); }
```

Comment:
I might call this `ExtractValue` instead of `GetValue` as a reminder to the caller that this leaves the stored value in a moved-from state instead of making a copy. (The `// copying` comment in the test case suggests that callers will often mistake this for returning a copy ;))